pkg/provider/vault/vault_test.go

```diff
@@ -327,36 +298,36 @@ MIICsTCCAZkCFEJJ4daz5sxkFlzq9n1djLEuG7bmMA0GCSqGSIb3DQEBCwUAMBMxETAPBgNVBAMMCHZh
 				err: fmt.Errorf(errGetKubeSecret, "vault-secret", errBoom),
 			},
 		},
-		"SuccessfulVaultStore": {
-			reason: "Should return a Vault provider successfully",
-			args: args{
-				store: makeSecretStore(),
-				kube: &test.MockClient{
-					MockGet: test.NewMockGetFn(nil, kubeMockWithSecretTokenAndServiceAcc),
-				},
-				newClientFunc: func(c *vault.Config) (Client, error) {
-					return &fake.VaultClient{
-						MockNewRequest: fake.NewMockNewRequestFn(&vault.Request{}),
-						MockRawRequestWithContext: fake.NewMockRawRequestWithContextFn(
-							newVaultTokenIDResponse("test-token"), nil, func(got *vault.Request) error {
-								kubeRole := makeValidSecretStore().Spec.Provider.Vault.Auth.Kubernetes.Role
-								want := kubeParameters(kubeRole, string(secretData))
-								if diff := cmp.Diff(want, got.Obj); diff != "" {
-									t.Errorf("RawRequestWithContext(...): -want, +got:\n%s", diff)
-								}
+		// "SuccessfulVaultStore": {
+		// 	reason: "Should return a Vault provider successfully",
+		// 	args: args{
+		// 		store: makeSecretStore(),
+		// 		kube: &test.MockClient{
+		// 			MockGet: test.NewMockGetFn(nil, kubeMockWithSecretTokenAndServiceAcc),
+		// 		},
+		// 		newClientFunc: func(c *vault.Config) (Client, error) {
+		// 			return &fake.VaultClient{
+		// 				MockNewRequest: fake.NewMockNewRequestFn(&vault.Request{}),
+		// 				MockRawRequestWithContext: fake.NewMockRawRequestWithContextFn(
+		// 					newVaultTokenIDResponse("test-token"), nil, func(got *vault.Request) error {
+		// 						kubeRole := makeValidSecretStore().Spec.Provider.Vault.Auth.Kubernetes.Role
+		// 						want := kubeParameters(kubeRole, string(secretData))
+		// 						if diff := cmp.Diff(want, got.Obj); diff != "" {
+		// 							t.Errorf("RawRequestWithContext(...): -want, +got:\n%s", diff)
+		// 						}
 
-								return nil
-							}),
-						MockSetToken:   fake.NewSetTokenFn(),
-						MockToken:      fake.NewTokenFn(""),
-						MockClearToken: fake.NewClearTokenFn(),
-					}, nil
-				},
-			},
-			want: want{
-				err: nil,
-			},
-		},
+		// 						return nil
+		// 					}),
+		// 				MockSetToken:   fake.NewSetTokenFn(),
+		// 				MockToken:      fake.NewTokenFn(""),
+		// 				MockClearToken: fake.NewClearTokenFn(),
+		// 			}, nil
+		// 		},
+		// 	},
+		// 	want: want{
+		// 		err: nil,
+		// 	},
+		// },
```

Comment:
I didn't fully understand what this test was testing. From what I could see, it was a use case no longer needed since we now use `kubernetesAuth` from vault.

Reply:
That is to ensure that the vault client passes the correct role and credentials are passed into this function:
```go
k, err := authkubernetes.NewKubernetesAuth("...")
v.auth.Login(ctx, k)
```

I think it's fine to remove it, this is tested e2e IIRC.